File: Sorting/Python/bubbleSort.py
def swap(array, i, j):
	temp = array[i]
	array[i] = array[j]
	array[j] =  temp

	return

def bubbleSort(array):
	#Number of elements.
	n = len(array)

	#Repeat n-1 times.
	for i in range(n):

		#Traverse the array from 0 to n-i-1
		#Swap if adyacent element is greater.
		for j in range(0, n - i -1):
			if(array[j] > array[j+1]):
				swap(array,j, j+1)
	return 

def printA(array):
	n = len(array)
	for  i in range(n):
		print("%d" %array[i]),
	print ("\n")
	return 

#Global variables
testArray = [1, 5, -1, 8, 2]

###################Main

#Array before sorting.
printA(testArray)

bubbleSort(testArray)

#Sorted array.
printA(testArray)

File: Sorting/C++/bubbleSort.cpp
#include <iostream>

using namespace std ;


void bubbleSort(int iArr[], int n){
	bool change =  true ;
	int i = 0, temp ; 

	//If there's no change the array is sorted. 
	while(change){
		change = false ;
		i++ ;
		for(int j = 0 ; j < n - i; j++){
			if(iArr[j] > iArr[j+1]){
				change = true ;

				//Swap the element.
				temp =  iArr[j] ;
				iArr[j] =  iArr[j+1] ;
				iArr[j+1] = temp ;
			}
		}
	}
}


void print(int iArr[], int n){
	for(int i = 0 ; i < n ; i ++){
		cout << iArr[i] << "\t" ;
	}
	cout << endl ;
}

int main(){
	int prueba[5] = {1, 8 , 7 , 2, 9} ;
	print(prueba, 5) ; 
	bubbleSort(prueba, 5) ;
	print(prueba, 5) ;

	return 0 ;
}
